src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,48 @@
 package core.basesyntax;
 
-class Pair {
+public class Pair<T,K> {
+    private T first;
+    private K second;
 
+    private Pair(T first, K second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public T getFirst() {
+        return first;
+    }
+
+    public K getSecond() {
+        return second;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        Pair<T,K> pair = (Pair) object;
+        return (first == pair.first
+                || pair != null && first.equals(pair.first))
+                && (second == pair.second
+                || pair != null
```

Comment:
do we need to check if pair == null  ?
i guess we did on 26 line

Reply:
Fixed